src/Interface.js

```
@@ -0,0 +1,236 @@
+ * @param {object} type
+ */
+export function context (value, type) {
+	switch (type) {
+		case 'svg':
+			return 'http://www.w3.org/2000/svg'
+		case 'math':
+			return 'http://www.w3.org/1998/Math/MathML'
+		case 'foreignObject':
+			return ''
+	}
+
+	return value
+}
+
+/**
+ * @param {object} parent
+ * @return {object}
+ */
+export function clear (parent) {
+	parent.textContent = ''
+}
+
+/**
+ * @param {object} parent
+ * @param {*} value
+ */
+export function content (parent, value) {
+	parent.nodeValue = value
+}
+
+/**
+ * @param {object} parent
+ * @param {object} element
+ */
+export function remove (parent, element) {
+	parent.removeChild(element)
+}
+
+/**
+ * @param {object} parent
+ * @param {object} element
+ */
+export function append (parent, element) {
+	parent.appendChild(element)
+}
+
+/**
+ * @param {object} parent
+ * @param {object} element
+ * @param {object} sibling
+ */
+export function insert (parent, element, sibling) {
+	parent.insertBefore(element, sibling)
+}
+
+/**
+ * @param {object} element
+ * @param {string} name
+ * @param {*} value
+ * @param {object} instance
+ */
+export function props (element, name, value, instance) {
+	if (name === 'style') {
+		stylesheet(name, value || false, instance)
+	} else {
+		switch (typeof value) {
+			case 'object': case 'function':
+				if (name.charCodeAt(0) === 111 && name.charCodeAt(1) === 110) {
+					return handler(element, name.substr(2).toLowerCase(), value, instance)
+				}
+		}
+
+		if (name in instance) {
+			switch (name) {
+				case 'width': case 'height':
+					break
+				default:
+					return property(name, value, instance)
+			}
+		}
+
+		attribute(name, value, instance)
+	}
+}
+
+/**
+ * @param {string} name
+ * @param {*} value
+ * @param {object} instance
+ */
+export function property (name, value, instance) {
+	try {
+		switch (instance[name] = value) {
+			case false: case null: case undefined:
+				return attribute(name, value, instance)
+		}
+	} catch (error) {
```

why swallowing error here?

---

Some svg properties on safari are readonly, and throw when set, so we fallback to setAttribute here.

---

should we `console.warn()` then?
```suggestion
	} catch (error) { // readonly properties (e.g. Safari svg)
```

---

It's not necessarily an author-centric error. but rather an artifact of the specific generic implementation as opposed to using a whitelist that might bloat the bundle size. So swallowing the error and re-trying with the non-setter variant `setAttribute`, is the intended behavior.